Microsoft.DotNet.ApiCompat failing with RegexMatchTimeoutException in

```
.packages/microsoft.dotnet.apicompat.task/8.0.100-alpha.1.22571.3/build/Microsoft.DotNet.ApiCompat.ValidateAssemblies.Common.targets(16,5): error MSB4018: (NETCORE_ENGINEERING_TELEMETRY=Build) The "Microsoft.DotNet.ApiCompat.Task.ValidateAssembliesTask" task failed unexpectedly.
System.Text.RegularExpressions.RegexMatchTimeoutException: The Regex engine has timed out while trying to match a pattern to an input string. This can occur for many reasons, including very large inputs or excessive backtracking caused by nested quantifiers, back-references and other factors.
   at System.Text.RegularExpressions.RegexRunner.<CheckTimeout>g__ThrowRegexTimeout|25_0()
   at Regex165_TryMatchAtCurrentPosition(RegexRunner, ReadOnlySpan`1)
   at Regex165_Scan(RegexRunner, ReadOnlySpan`1)
   at System.Text.RegularExpressions.Regex.RunAllMatchesWithCallback[TState](String inputString, ReadOnlySpan`1 inputSpan, Int32 startat, TState& state, MatchCallback`1 callback, RegexRunnerMode mode, Boolean reuseMatchObject)
   at System.Text.RegularExpressions.RegexReplacement.Replace(Regex regex, String input, Int32 count, Int32 startat)
   at Microsoft.DotNet.ApiCompat.RegexStringTransformer.Transform(String input) in /_/src/ApiCompat/Microsoft.DotNet.ApiCompat.Shared/RegexStringTransformer.cs:line 51
```

## Build Information
Build: https://dev.azure.com/dnceng-public/cbb18261-c48f-4abb-8651-8cdcb5474649/_build/results?buildId=157087
Build error leg or test failing: Build / Libraries Build osx arm64 Debug / Restore and Build Product
Pull request: https://github.com/dotnet/runtime/pull/81510

## Error Message
Fill the error message using [known issues guidance](https://github.com/dotnet/arcade/blob/main/Documentation/Projects/Build%20Analysis/KnownIssues.md#how-to-fill-out-a-known-issue-error-message-section).

```json
{
  "ErrorMessage": "Microsoft.DotNet.ApiCompat.RegexStringTransformer.Transform",
  "BuildRetry": true,
  "ErrorPattern": "",
  "ExcludeConsoleLog": false
}
```

<!--Known issue error report start -->
### Report
|Build|Definition|Step Name|Console log|Pull Request|
|---|---|---|---|---|
|[157151](https://dev.azure.com/dnceng-public/public/_build/results?buildId=157151)|dotnet/runtime|Restore and Build Product|[Log](https://dev.azure.com/dnceng-public/cbb18261-c48f-4abb-8651-8cdcb5474649/_apis/build/builds/157151/logs/1951)|dotnet/runtime#81535|
|[157087](https://dev.azure.com/dnceng-public/public/_build/results?buildId=157087)|dotnet/runtime|Restore and Build Product|[Log](https://dev.azure.com/dnceng-public/cbb18261-c48f-4abb-8651-8cdcb5474649/_apis/build/builds/157087/logs/531)|dotnet/runtime#81510|
#### Summary
|24-Hour Hit Count|7-Day Hit Count|1-Month Count|
|---|---|---|
|0|2|2|
<!--Known issue error report end -->

--- a/src/ApiCompat/Microsoft.DotNet.ApiCompat.Shared/RegexStringTransformer.cs
+++ b/src/ApiCompat/Microsoft.DotNet.ApiCompat.Shared/RegexStringTransformer.cs
@@ -12,8 +12,6 @@ namespace Microsoft.DotNet.ApiCompat
     internal class RegexStringTransformer
     {
         private readonly (Regex Regex, string ReplacementString)[] _patterns;
-        // Define a timeout for regex matches to guard against malicious user inputs.
-        private static readonly TimeSpan s_regexTimeout = TimeSpan.FromSeconds(2);
 
         /// <summary>
         /// Initializes the regex string transformer with a given capture group and replacement patterns.
@@ -34,7 +32,7 @@ namespace Microsoft.DotNet.ApiCompat
             _patterns = new (Regex Regex, string ReplacementString)[rawPatterns.Length];
             for (int i = 0; i < rawPatterns.Length; i++)
             {
-                _patterns[i] = (new Regex(rawPatterns[i].CaptureGroupPattern, RegexOptions.Compiled, s_regexTimeout), rawPatterns[i].ReplacementString);
+                _patterns[i] = (new Regex(rawPatterns[i].CaptureGroupPattern, RegexOptions.Compiled), rawPatterns[i].ReplacementString);
             }
         }
 

--- a/src/Tests/Microsoft.DotNet.ApiCompat.Tests/RegexStringTransformerTests.cs
+++ b/src/Tests/Microsoft.DotNet.ApiCompat.Tests/RegexStringTransformerTests.cs
@@ -74,15 +74,5 @@ namespace Microsoft.DotNet.ApiCompatibility.Tests
 
             Assert.Equal("runtimes/android/lib/net7.0/System.Linq.dll", output);
         }
-
-        [Fact]
-        public void Transform_SinglePatternWithBacktracking_ThrowsRegexMatchTimeoutException()
-        {
-            const string TransformInput = "An input string that takes a very very very very very very very very very very very long time!";
-
-            RegexStringTransformer regexStringTransformer = new(@"^(\w+\s?)*$", "lib");
-
-            Assert.Throws<RegexMatchTimeoutException>(() => regexStringTransformer.Transform(TransformInput));
-        }
     }
 }

